### PR TITLE
Test/rows 100 coverage

### LINKED ISSUE #195: Create unittests for `switchmap/server/db/misc/rows.py`
We will need unittests done for all methods, classes and functions found in this file.

### PR Acceptance Criteria

1. When complete this file must show **100%** coverage when merged into the code base.
1. [The current code coverage for the file can be found here](https://app.codecov.io/gh/PalisadoesFoundation/switchmap-ng/tree/develop/?displayType=list)
1. CodeRabbit must approve all your changes to the test file(s) and the file(s) under test
1. The PR will show a report for the code coverage for the file you have added. You can use that as a guide
1. No functionality must be removed to achieve this goal
1. All tests must be valid and truly test the code base
1. The tests must be placed in the equivalent folder under `tests/switchmap_` therefore:
    1. a file named `switchmap/x/y/z/file.py` must have a test file named `tests/switchmap_/x/y/z/test_file.py`
    1. file `tests/switchmap_/x/y/z/test_file.py` must have 755 permissions so it can be easily executable
    1. take guidance on the formatting of the file from other `tests/switchmap_/x/y/z/test_*.py` files or others along the path or in adjacent directories
    
### Other Criteria

1. If this file isn't referenced in the code base then remove it. We don't want tests run on unused code

### Potential internship candidates

Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Abhi-MS**
**What kind of change does this PR introduce?**

Added unit tests for coverage on the rows.py file.

**Issue Number:**

Fixes #195 

**Snapshots/Videos:**

image

**If relevant, did you update the documentation?**

Not relevant

**Summary**

This PR introduces unit tests for all functions in the rows.py file to achieve 100% test coverage.
The tests ensure that the following functions correctly convert database rows to tuples:

- device()
- root()
- event()
- l1interface()
- mac()
- macip()
- macport()
- oui()
- vlan()
- vlanport()
- zone()
- ip()
- ipport()

Additionally, edge cases where the input is None are tested to ensure that each function properly raises exceptions when provided with invalid input.

The tests verify the following:
- Correct tuple conversion for each function.
- Correct tuple length and data type.
- Correct value in the tuple for each entry in the row
- Handling of invalid (None) inputs with appropriate exceptions.

**Does this PR introduce a breaking change?**

No

### CodeRabbit AI Review
- [ ] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [ ] I have implemented or provided justification for each non-critical suggestion
- [ ] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [ ] I have written tests for all new changes/features
- [ ] I have verified that test coverage meets or exceeds 95%
- [ ] I have run the test suite locally and all tests pass

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/switchmap-ng/blob/main/CONTRIBUTING.md)?**

Yes

## Summary by CodeRabbit

- **Tests**
  - Introduced a new automated test module to verify that data conversion processes work as expected.
  - The tests ensure accurate formatting and proper error handling when invalid inputs are encountered, contributing to overall system reliability.
  - Added comprehensive unit tests for various entity conversions, including edge case validations.  
  - Improved error detection by refining pattern matching for numeric codes in function argument parsing.

### CODE REVIEW

File: tests/switchmap_/server/db/misc/test_rows.py [hunk 1]
```diff
@@ -0,0 +1,660 @@
...
+
+    def test_ip_edge_none(self):
+        """Edge Case: Test ip() with None input."""
+        with self.assertRaises(Exception):
+            rows.ip(None)
+
+    def test_ipport_edge_none(self):
+        """Edge Case: Test ipport() with None input."""
+        with self.assertRaises(Exception):
+            rows.ipport(None)
```
[the remark this answers is not in the document]
Comment (Abhi-MS) [reply tests/switchmap_/server/db/misc/test_rows.py:656]: assertRaises(Exception) is changed to assertRaises(AttributeError)

**Comment by palisadoes** [comment]
1. Please make coderabbit.ai approves your work.
2. Resolve any conflicting files
3. Make sure all tests pass and are valid.
4. Ensure the test code coverage for your patch reaches as close to **100%** as possible.

**Comment by palisadoes** [comment]
1. There is a check to make sure the error codes in log.debug don't overlap.
2. Please check again

**Comment by Abhi-MS** [comment]
@palisadoes Thank you for the feedback. I am having difficulty pinpointing where this error is coming from; it is not directly from the test file.
image

**Comment by Abhi-MS** [comment]
@coderabbitai full review

**Comment by Abhi-MS** [comment]
@palisadoes This PR is now passing all checks. The issue was due to the number 1 in the following line being incorrectly interpreted as an error code by the error_code_report.py script:
image

This behavior stems from the following function, which could be improved to avoid such false positives in the future:
image

Would you like me to create an issue addressing this?

**Comment by palisadoes** [comment]
> @palisadoes This PR is now passing all checks. The issue was due to the number 1 in the following line being incorrectly interpreted as an error code by the error_code_report.py script: image
> 
> This behavior stems from the following function, which could be improved to avoid such false positives in the future: image
> 
> Would you like me to create an issue addressing this?

Fix it in this PR

**Comment by Abhi-MS** [comment]
> > @palisadoes This PR is now passing all checks. The issue was due to the number 1 in the following line being incorrectly interpreted as an error code by the error_code_report.py script: image
> > This behavior stems from the following function, which could be improved to avoid such false positives in the future: image
> > Would you like me to create an issue addressing this?
> 
> Fix it in this PR

I’ve fixed errors.py by adding a comma to the regex pattern, since all the detected lines follow that structure (as shown in the screenshot). This should prevent cases like 1 in L1Interface from being mistakenly identified as error codes.
image

**Review by palisadoes — APPROVED** [review]
